frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts

```diff
@@ -0,0 +1,140 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { UserProfilePageComponent } from './user-profile-page.component';
+import { RouterTestingModule } from '@angular/router/testing';
+import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
+import { of, Subject } from 'rxjs';
+import { UserFacadeService } from './store/user/user.facade';
+import { takeUntil } from 'rxjs/operators';
+import { EntityWrapper } from '../auth/models/entity-wraper';
+import { EntityStatus } from '../auth/models/entity-status';
+import { User } from '../auth/models/user';
+
+describe('UserProfileComponent', () => {
+	const user: EntityWrapper<User> = {
+		status: EntityStatus.Success,
+		value: {
+			email: 'test@test.by',
+			password: 'test',
+			firstName: 'test',
+			secondName: 'test',
+			mobilePhone: 'test',
+			id: 'test',
+			isAdmin: false,
+		},
+	};
+
+	const updateUser: any = {
+		firstName: 'test',
+		secondName: 'test',
+		mobilePhone: 'test',
+		_id: 'test',
+	};
+
+	let component: UserProfilePageComponent;
+	let destroy$: Subject<boolean>;
+	let userForm: User;
+	let fixture: ComponentFixture<UserProfilePageComponent>;
+	const mockUserFacadeService: any = {
+		get user$(): any {
+			return of(user);
+		},
+		getUserFromAuth(): any {
+			return {};
+		},
+		updateUser: jasmine.createSpy('updateUser'),
+	};
+
+	beforeEach(async(() => {
+		TestBed.configureTestingModule({
+			declarations: [UserProfilePageComponent],
+			imports: [ReactiveFormsModule, RouterTestingModule],
+			providers: [{ provide: UserFacadeService, useValue: mockUserFacadeService }, { provide: FormBuilder }],
+		}).compileComponents();
+	}));
+
+	beforeEach(() => {
+		fixture = TestBed.createComponent(UserProfilePageComponent);
+		component = fixture.componentInstance;
+		user.status = EntityStatus.Success;
+		userForm = user.value;
+		fixture.detectChanges();
+	});
+
+	it('should create', () => {
+		expect(component).toBeTruthy();
+	});
+
+	describe('submit', () => {
+		beforeEach(() => {
+			component.updateUserInfo();
+		});
+		it('should redirect', () => {
+			expect(mockUserFacadeService.updateUser).toHaveBeenCalledWith(updateUser);
+		});
+	});
+
+	describe('when ngOnInit', () => {
+		beforeEach(() => {
+			destroy$ = new Subject<boolean>();
+		});
+
+		describe('get auth user', () => {
+			beforeEach(() => {
+				mockUserFacadeService.user$.pipe(takeUntil(destroy$)).subscribe((users: any) => {
+					userForm = users.value;
+				});
+				spyOn<any>(component, 'setValuesToForm').and.stub();
```

Comment:
you can use non generic spyOn here

Reply:
its one of the way to do test of private method, 
https://stackoverflow.com/questions/17885635/private-method-unit-testing-with-jasmine/38144942